fast_matched_filter/fast_matched_filter.py

```diff
@@ -194,10 +211,18 @@ def matched_filter(
         print("Template and data dimensions are not compatible!")
         return
 
-    n_samples_template = templates.shape[-1]
-    if templates.shape != (n_templates, n_stations, n_components, n_samples_template):
+    if n_samples_template is None:
+        n_samples_template = templates.shape[-1]
+        max_samples_template = n_samples_template
+        variable_template_length = False
+    else:
```

Comment:
You might want to check that `n_samples_template` is the right shape. Someone could get the wrong impression and set `n_samples_template=200` for a fixed length template for example.

Reply:
Fair point. I had an assert further down, but I just moved it up here.